### DISCUSSION TITLE
Implementing windows authentication while connecting sql datasource to store Quartz jobs

### DISCUSSION THREAD

**Discussion by bkdprakash** [quartznet/quartznet#1583] [Q&A]
we are using QuartzNet to implement background tasks in web application. As per our standards, we have implemented sql windows authentication in connectionstring (web.config) file.

In the same way, we have to implement windows authentication in Quartz connection string also. But, when we remove 'User Id' & 'Password' from Quartz connection string, the jobs are not storing into Quartz database. 

Any way to handle this?

**Comment by lahma** [comment]
I think you should enable logging and see what connection errors Quartz is getting. There's not much Quartz itself can do as it's connection string / database server setup specific .

**Comment by bkdprakash** [comment]
we are getting following error when we remove User Id & Password.

Failed to obtain DB connection from data source 'QRTZDS': System.Data.SqlClient.SqlException (0x80131904): Login failed for user ''.     at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)     at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)     at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)     at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)     at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)     at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)     at System.Data.SqlClient.SqlConnection.Open()     at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection() in c:\Program Files (x86)\Jenkins\workspace\Quartz.NET\src\Quartz\Impl\AdoJobStore\JobStoreSupport.cs:line 367  ClientConnectionId:54b01106-ed57-43d0-93de-44b4244ddefc  Error Number:18456,State:1,Class:14

**Reply by lahma** [reply]
There's probably something wrong with your connection string or the user account running the process hosting Quartz.

**Reply by jafin** [reply]
@bkdprakash . yeah definately check your connection string, or the context your app in running in.

I just ran a simple test using
```c#
x.UseSqlServer("Server=localhost;Database=QuartzTest;Trusted_Connection=True;TrustServerCertificate=True");
```
and Quartz authenticated successfully to MSSQL.
